pkg/service/generateConfig/generateConfig.go

```diff
@@ -0,0 +1,97 @@
+package generateConfig
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+
+	"go.keploy.io/server/utils"
+
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+var Emoji = "\U0001F430" + " Keploy:"
+
+type generatorConfig struct {
+	logger *zap.Logger
+	mutex  sync.Mutex
+}
+
+func NewGeneratorConfig(logger *zap.Logger) GeneratorConfig {
+	return &generatorConfig {
+		logger: logger,
+		mutex:  sync.Mutex{},
+	}
+}
+
+var config = `
+record:
+  path: ""
+  # mandatory
+  command: ""
+  proxyport: 0
+  containerName: ""
+  networkName: ""
+  delay: 5
+  passThroughPorts: []
+test:
+  path: ""
+  # mandatory
+  command: ""
+  proxyport: 0
+  containerName: ""
+  networkName: ""
+  testSets: []
+  globalNoise: |-
+    {
+      "global": {
+        "body": {},
+        "header": {}
+      },
+      "test-sets": {
+        "test-set-name": {
+          "body": {},
+          "header": {}
+        }
+      }
+    }
+  delay: 5
+  apiTimeout: 5
+  passThroughPorts: []
+`
+
+func (g *generatorConfig) GenerateConfig(path string) {
+	var node yaml.Node
+
+	data := []byte(config)
+
+	if err := yaml.Unmarshal(data, &node); err != nil {
+		g.logger.Fatal("Unmarshalling failed %s", zap.Error(err))
+	}
+
+	results, err := yaml.Marshal(node.Content[0])
+	if err != nil {
+		g.logger.Fatal("Failed to marshal the config", zap.Error(err))
+	}
+
+	filePath := filepath.Join(path, "keploy-config.yaml")
+
+	if utils.CheckFileExists(filePath) {
+		override, err := utils.AskForConfirmation("Config file already exists. Do you want to override it?")
```

Comment:
there is a [prerunE](https://github.com/spf13/cobra/issues/1007) but it would create redundancy as we would need to read the configPath flag again. I have done it in a conventional way !

Reply:
Why do we read configPath again didn't get you ..? prerunE is some kind of function which is run before running its major function right . You can show example regarding this redundancy you mentioned

Reply:
PrerunE must read config path to decide whether a file existence found at the location pointed by configpath

Reply:
Doesn't prerunE take flags provided in the command..?

```
var RootCmd = &cobra.Command{
    Use: "test",
    Run: func(c *cobra.Command, args []string) {
        fmt.Println("Run")
    },
    PreRun: func(c *cobra.Command, args []string) {
        fmt.Println("PreRun")
    },
```
In the example above args are sent into both the functions right run and prerun? I though prerun is something like this . Just check once and correct me if i am wrong.

Reply:
args is populated only with there are arguments passed before any flags. like if we want to populate the args, we can write a command like:
`keploy generate-config "/home/"`
args will get populated with ["/home/"]

if we issue a command:
`keploy generate-config --path "/home"
the args would remain empty as there were no additional arguments passed before the flags.

Reply:
Got it.